### PR TITLE
containers: regroup and add Alpine images (31 MB)

### PR DESCRIPTION
**Pull request by karalabe**
This PR moves the existing docker images one level deeper to cater for additional container types (vagrant specifically now) without polluting the namespace with them. Additionally the PR tags the current containers with `-ubuntu` and adds a new set of images based on the Alpine linux distribution. 

The Ubuntu images are needed if someone wants to extend our images and build arbitrary stuff on top of them (e.g. build/ci servers), but Alpine is getting quite a lot of PR lately as being one of the smallest distros (5MB image size), and is being endorsed both by CoreOS as well as Docker is moving their stuff over to Alpine. These two reasons + that an Alpine based Geth is 31MB in size is imho enough to warrant supporting it along the 2-3x larger Ubuntu version.

@caktux I'm not sure how this reorg affects your docker images and/or the build servers, so could you maybe give your two cents for this PR? Thanks :) I would also appreciate if you could verify if the images are correct or if something should be done differently.

### CODE REVIEW

**Comment by robotally** [comment]
| Vote | Count | Reviewers |
| :---: | :---: | :---: |
| :+1: | 0 |  |
| :-1: | 0 |  |

_Updated: Sat Jan 30 19:04:29 UTC 2016_

**Comment by caktux** [comment]
They won't affect the builds, and they're solid. :+1: :+1: I'll rig the automated builds on docker hub for these puppies.

File: containers/docker/master-alpine/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM alpine:3.3
+
+RUN \
+  apk add --update go git make gcc musl-dev gmp-dev && \
```
Comment (obscuren) [inline containers/docker/master-alpine/Dockerfile]: May reomove GMP

File: containers/docker/master-alpine/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  git clone https://github.com/ethereum/go-ethereum && \
+  (cd go-ethereum && make geth)                     && \
+  cp go-ethereum/build/bin/geth /geth               && \
+  apk del go git make gcc musl-dev gmp-dev          && \
```
Comment (obscuren) [inline containers/docker/master-alpine/Dockerfile]: Same here

**Comment by karalabe** [comment]
@caktux Merged, please check that docker hub correctly assembles the new images :)

**Pull request merged**